openedx/core/djangoapps/content/block_structure/migrations/0005_trim_leading_slashes_in_data_path.py

```diff
@@ -0,0 +1,46 @@
+"""
+Data migration to convert absolute paths in block_structure.data to be relative.
+
+This has only been tested with MySQL, though it should also work for Postgres as
+well. This is necessary to manually correct absolute paths in the "data" field
```

Comment:
I don't think this SQL will work on Postgres, as the equivalent of `length()` on PostgreSQL is `char_length()`:
https://www.postgresql.org/docs/9.1/functions-string.html

Also, sqlite doesn't have a `RIGHT()` function, so it won't work there.

I don't know how cross-DB this migration needs to be - or what our best practice has been for raw SQL in the past.

Reply:
`length` is supported as well, though I don't know exactly what the difference in between it and `char_length` is:

```
psql (13.3)
Type "help" for help.

daveormsbee=# select length('Hello World!');
 length 
--------
     12
(1 row)

daveormsbee=# select right('/courses/', length('/courses/') - 1);
  right   
----------
 courses/
(1 row)
```

Reply:
As for SQLite–I think the only thing we have to worry about are tests? And that seems to work because we skip migrations?

Reply:
Hmm - guess one is an alias for the other in pgSQL? Agreed on SQLite - so 👍🏼